app/utils/discogs.ts

```diff
@@ -1,25 +1,29 @@
 import { SearchResponseSchema } from "./discogs.schemas";
 
-const discogs = {
-  search: async () => {
+type searchResult = {
+  results: [{ year?: string }];
+};
 
+const discogs = {
+  search: async (search: string): Promise<searchResult> => {
```

Comment:
woops, guess my comments didn't post here. You can just use your zod schema here, no need for us to make another type:
```suggestion
  search: async (search: string): Promise<z.infer<typeof SearchResponseSchema>> => {
```
you can just remove `searchResult`